src/twisted/conch/test/sk_dummy.py

```diff
@@ -0,0 +1,185 @@
+
+
+class SKAlgorithm(Enum):
+    """Supported Security Key algorithms."""
+
+    ECDSA = 1
+    ED25519 = 2
+
+
+@dataclass
+class EnrollResponse:
+    """Structured response for an enrollment operation."""
+
+    public_key: bytes
+    key_handle: bytes
+    signature: bytes
+    attestation_cert: Optional[bytes] = None
+    flags: int = 0
+
+
+@dataclass
+class SignResponse:
+    """Structured response for a signing operation."""
+
+    flags: int
+    counter: int
+    signature_r: bytes
+    signature_s: Optional[bytes]  # None for Ed25519 algorithm
+
+
+class SKError(Exception):
+    """Custom exception for Security Key errors."""
+
+    pass
+
+
+class DummySK:
+    """
+    A dummy software-based implementation of the OpenSSH Security Key API.
+
+    This class mimics the behavior of the `sk-dummy.c` library for testing
+    purposes. It simulates a hardware key entirely in software.
+
+    WARNING: This is insecure. The 'key_handle' produced by enroll() is the
+    actual private key. Do NOT use this for anything other than testing.
+    """
+
+    def enroll(
+        self,
+        alg: SKAlgorithm,
+        challenge: bytes,
+        application: str,
+        flags: int,
+        pin: Optional[str] = None,
+    ) -> EnrollResponse:
+        """
+        Simulates the enrollment of a new security key credential.
+
+        Args:
+            alg: The algorithm for the new key (ECDSA or ED25519).
+            challenge: The challenge data from the server (unused in dummy).
+            application: The application identifier (e.g., "ssh:").
+            flags: User presence/verification flags.
+            pin: The user's PIN (unused in dummy).
+
+        Returns:
+            An EnrollResponse containing the public key and a key_handle.
+            The key_handle is the insecurely stored private key.
+        """
+        if alg == SKAlgorithm.ECDSA:
+            ec_private_key = ec.generate_private_key(ec.SECP256R1())
+
+            public_key = ec_private_key.public_key().public_bytes(
+                encoding=serialization.Encoding.X962,
+                format=serialization.PublicFormat.UncompressedPoint,
+            )
+
+            key_handle = ec_private_key.private_bytes(
+                encoding=serialization.Encoding.PEM,
+                format=serialization.PrivateFormat.PKCS8,
+                encryption_algorithm=serialization.NoEncryption(),
+            )
+        elif alg == SKAlgorithm.ED25519:
+            ed_private_key = ed25519.Ed25519PrivateKey.generate()
+
+            public_key = ed_private_key.public_key().public_bytes(
+                encoding=serialization.Encoding.Raw,
+                format=serialization.PublicFormat.Raw,
+            )
+
+            key_handle = ed_private_key.private_bytes(
+                encoding=serialization.Encoding.Raw,
+                format=serialization.PrivateFormat.Raw,
+                encryption_algorithm=serialization.NoEncryption(),
+            )
+
```

Comment:
Maybe have something like this to make the coverage happy.
We explicitly inform that we only support these 2 types


```suggestion
            )
        else:  # pragma:r no cover
            raise AssertionError('Unsuported algorithm.')

```